File: 117. Update Task Status Page/TaskManager-Angular/src/app/models/task.ts
import { Project } from './project';

export class Task
{
    taskID: number;
    taskName: string;
    description: string;
    createdOn: string;
    projectID: number;
    createdBy: string;
    assignedTo: string;
    taskPriorityID: number;
    lastUpdatedOn: number;
    currentStatus: number;
    currentTaskStatusID: number;

    project: Project;
    createdByUser: any;
    assignedToUser: any;
    taskStatusDetails: any;

    constructor()
    {
        this.taskID = null;
        this.taskName = null;
        this.description = null;
        this.createdOn = null;
        this.projectID = null;
        this.createdBy = null;
        this.assignedTo = null;
        this.taskPriorityID = null;
        this.lastUpdatedOn = null;
        this.currentStatus = null;

        this.project = null;
        this.createdByUser = null;
        this.assignedToUser = null;
        this.taskStatusDetails = null;
        this.taskStatusDetails = null;
        this.currentTaskStatusID = null;
    }
}

File: Old/18. Final Course Project/11. Applying Contextual Classes to Tasks/MvcTaskManager-AspNetCore2.1/MvcTaskManager/Models/Task.cs
using MvcTaskManager.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MvcTaskManager.Models
{
    public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TaskID { get; set; } //Primary Key for Tasks table
        public string TaskName { get; set; } //Name / title of the task
        public string Description { get; set; } //Description of the task
        public DateTime CreatedOn { get; set; } //Date and time of task creation
        public int ProjectID { get; set; } //FK, refers to "Projects" table
        public string CreatedBy { get; set; } //FK, refers to "AspNetUsers" table, indicating who created the task
        public string AssignedTo { get; set; } //FK, refers to "AspNetUsers" table, indicating to whom the task is assigned
        public int TaskPriorityID { get; set; } //FK, refers to "TaskPriorities" table
        public DateTime LastUpdatedOn { get; set; } //Date and time of task last updation
        public string CurrentStatus { get; set; } //Current status name of the task
        public int CurrentTaskStatusID { get; set; } //Indicating ID of current task status

        [NotMapped]
        public string CreatedOnString { get; set; }
        [NotMapped]
        public string LastUpdatedOnString { get; set; }

        [ForeignKey("ProjectID")]
        public virtual Project Project { get; set; }

        [ForeignKey("CreatedBy")]
        public virtual ApplicationUser CreatedByUser { get; set; }

        [ForeignKey("AssignedTo")]
        public virtual ApplicationUser AssignedToUser { get; set; }

        [ForeignKey("TaskPriorityID")]
        public virtual TaskPriority TaskPriority { get; set; }

        public virtual ICollection<TaskStatusDetail> TaskStatusDetails { get; set; }
    }

    public class GroupedTask
    {
        public string TaskStatusName { get; set; }
        public List<Task> Tasks { get; set; }
    }
}


